code/api/src/migrations/20210325155429-5-userproducts.js.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+module.exports = {
```

Comment:
Alright so you're exporting an object that has two methods: `up`, and `down`.

These take in two parameters: `queryInterface` and `Sequelize`.

It returns whatever is the output of a `createTable()` method call from the `queryInterface`. I assume it makes a table with the key of the first argument `'userProducts'` and the properties of the second argument `'id, userId'...` and types them as denoted by the `Sequelize.TYPE`.

My understanding is that Sequelize is used to connect to the postgres database via javascript (in addition/in place of sql commands?). It's an ORM (object relational mapping) library that lets us change from the way data is stored in the psql tables into objects with properties that we can use in JS.

Reply:
Correct, Sequelize is an ORM that is turning JavaScript into SQL to give to the database, and turning SQL into JavaScript when returned from the database.